src/components/NewSubtask.jsx

```diff
@@ -0,0 +1,90 @@
+import React, { useState, useContext } from "react";
+import AddButton from "./global/AddButton";
+import NewSubtaskModal from "./global/Modal";
+import IconButton from "./global/IconButton";
+import Button from "./global/Button";
+import TextInput from "./global/TextInput";
+import { createSubtask } from "../services/databaseService";
+import { AuthContext } from "../services/providers/authProvider";
+
+export default function NewSubtask(props) {
+  const { taskId } = props;
+
+  const [display, setDisplay] = useState(false);
+  const [name, setName] = useState("");
+  const [error, setError] = useState("");
+  const { currentUser } = useContext(AuthContext);
+
+  // reset the values of the new subtask modal so fields are empty when it is reopened.
+  function resetValues() {
+    setName("");
+    setError("");
+  }
+
+  // closes the new subtask modal
+  function handleCancelNewSubtask() {
+    setDisplay(false);
+    resetValues();
+  }
+
+  // makes a request to the database to persist the new subtask.
+  async function handleAddNewSubtask() {
+    if (!name) {
+      setError("Please enter a subtask name.");
+      return;
+    }
+
+    createSubtask(name, taskId, currentUser.uid);
+
+    setDisplay(false);
+    resetValues();
+  }
+
+  /**
+   * called for any key press while user is focused on the modal.
+   * if the user presses "Enter", the supplied clickHandler function is called.
+   * @param {*} event Keypress event
+   */
+  function handleKeyPress(event) {
+    if (event.key === "Enter") {
+      handleAddNewSubtask();
+    }
+  }
+
+  return (
+    <div className="new-subtask-container">
+      <AddButton onClick={() => setDisplay(true)} />
+      <NewSubtaskModal
+        dismissOnClickOutside
+        onCancel={handleCancelNewSubtask}
+        show={display}
+        handleKeyPress={handleKeyPress}
+      >
+        <TextInput
+          label="Subtask Title"
+          textValue={name}
+          onChangeHandler={setName}
+        />
+        <div className="hBox">
```

Comment:
I think you should try and keep the styling consistent with their conventions. I think that the current class names of the file that you took inspiration from isn't quite up to their standard. You should check out [BEM](https://yandex.com/dev/bem/) for some more info

Reply:
Thanks I'll change this!